app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -0,0 +1,126 @@
+import { useState, useMemo } from 'react'
+import { Alert, CalendarMonth, FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core'
+
+import { createReactWrapper } from 'utilities/createReactWrapper'
+import type { IRecord } from 'utilities/patternfly-utils'
+
+import type { FunctionComponent, FormEvent } from 'react'
+
+import './ExpirationDatePicker.scss'
+
+interface ExpirationItem extends IRecord {
+  id: number;
+  name: string;
+  period: number; // In seconds
+}
+
+const collection: ExpirationItem[] = [
+  { id: 1, name: '7 days', period: 7 },
+  { id: 2, name: '30 days', period: 30 },
+  { id: 3, name: '60 days', period: 60 },
+  { id: 4, name: '90 days', period: 90 },
+  { id: 5, name: 'Custom...', period: 0 },
+  { id: 6, name: 'No expiration', period: 0 }
+]
+
+const dayMs = 60 * 60 * 24 * 1000
+const msExp = /\.\d{3}Z$/
+
+interface Props {
+  id: string;
+  label: string | null;
+}
+
+const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label }) => {
+  const [selectedItem, setSelectedItem] = useState(collection[0])
+  const [pickedDate, setPickedDate] = useState(new Date())
+  const fieldName = `human_${id}`
+  const fieldLabel = label ?? 'Expires in'
+
+  const fieldDate = useMemo(() => {
+    if (selectedItem.period === 0) return null
+
+    return new Date(new Date().getTime() + selectedItem.period * dayMs)
+  }, [selectedItem])
+
+  const fieldHint = useMemo(() => {
+    if (!fieldDate) return
+
+    const date = new Date(fieldDate)
+    date.setHours(0, 0, 0, 0)
+
+    return `The token will expire on ${date.toLocaleDateString()}`
+  }, [fieldDate])
+
+  const dateValue = useMemo(() => {
+    let value = ''
+
+    if (fieldDate) {
+      value = fieldDate.toISOString().replace(msExp, 'Z')
+    } else if (selectedItem.id === 5 ) {
+      value = pickedDate.toISOString().replace(msExp, 'Z')
+    }
+
+    return value
```

Comment:
```suggestion
    if (fieldDate) {
      value = fieldDate
    } else if (selectedItem.id === 5 ) {
      value = pickedDate
    }

    return value.toISOString().replace(msExp, 'Z')
```

On the other hand - maybe we can accept both formats on the backend by using `DateTime.parse`? It seems to work both with milliseconds and no milliseconds.

Reply:
I'll explore this possibility, it might work

Reply:
https://github.com/3scale/porta/pull/3943/commits/d2f74118a5ef06c481056d65de34b645ae011065